### PR TITLE
Added Content type sidebar UI location

### PR DESCRIPTION
**Pull request by praveen-mohan-cs**

### CODE REVIEW

**Review by rijil-tr — CHANGES_REQUESTED** [review]

File: src/ContentTypeSidebarWidget.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * Executes the provided callback function every time a content type is saved.
+     * @param {function} callback - The function to be called when a content type is saved.
+     * @param {ContentType} arg0 - The content type data passed as an argument to the callback function when a content type is saved.
+     */
+    onSave(callback: (arg0: ContentType) => void) {
+        const contentTypeObj = this;
+        if (callback && typeof callback === "function") {
+            contentTypeObj._emitter.on(
```
[the remark this answers is not in the document]
Comment (praveen-mohan-cs) [reply src/ContentTypeSidebarWidget.ts:56]: Done. Added the code for that.

File: src/uiLocation.ts [hunk 2]
```diff
@@ -345,6 +358,15 @@ class UiLocation {
                         { data: event.data.data },
                     ]);
                 }
+
+                if (event.data.name === "contentTypeSave") {
+                    emitter.emitEvent("contentTypeSave", [
+                        { data: event.data.data },
+                    ]);
+                    emitter.emitEvent("updateFields", [
```
[the remark this answers is not in the document]
Comment (praveen-mohan-cs) [reply src/uiLocation.ts]: It's not required. Misunderstood the usecase here. I have removed that event.

**Comment by praveen-mohan-cs** [comment]
@rijil-tr Addressed your comments. Can you please take a look once and approve ?

File: src/ContentTypeSidebarWidget.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import EventEmitter from "wolfy87-eventemitter";
+import { IContentTypeSidebarInitData } from "./types";
+import postRobot from "post-robot";
+import { ContentType } from "./types/stack.types";
+import { GenericObjectType } from "./types/common.types";
```
Comment (abhishek305) [inline src/ContentTypeSidebarWidget.ts:5]: follow proper import approach

File: src/ContentTypeSidebarWidget.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+     * Executes the provided callback function every time a content type is saved.
+     * @param {function} callback - The function to be called when a content type is saved.
+     * @param {ContentType} arg0 - The content type data passed as an argument to the callback function when a content type is saved.
+     */
+    onSave(callback: (arg0: ContentType) => void) {
+        const contentTypeObj = this;
+        if (callback && typeof callback === "function") {
+            contentTypeObj._emitter.on(
+                "contentTypeSave",
+                (event: { data: any }) => {
```
Comment (abhishek305) [inline src/ContentTypeSidebarWidget.ts:58]: use generic object type instead of any

File: src/ContentTypeSidebarWidget.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    getData(): ContentType {
+        return this.currentContentType;
+    }
+
+    /**
+     * Executes the provided callback function every time a content type is saved.
+     * @param {function} callback - The function to be called when a content type is saved.
+     * @param {ContentType} arg0 - The content type data passed as an argument to the callback function when a content type is saved.
+     */
+    onSave(callback: (arg0: ContentType) => void) {
```
Comment (abhishek305) [inline src/ContentTypeSidebarWidget.ts:53]: why arg0 ?

**Review by abhishek305 — COMMENTED** [review]
@praveen-mohan-cs minor comments . rest looks good

File: src/ContentTypeSidebarWidget.ts (hunk 3 above)
Comment (praveen-mohan-cs) [reply src/ContentTypeSidebarWidget.ts:5]: addressed it

File: src/ContentTypeSidebarWidget.ts (hunk 4 above)
Comment (praveen-mohan-cs) [reply src/ContentTypeSidebarWidget.ts]: done

File: src/ContentTypeSidebarWidget.ts (hunk 5 above)
Comment (praveen-mohan-cs) [reply src/ContentTypeSidebarWidget.ts:53]: this is how the arguments are named in entry and asset sidebar, just followed the same here

**Review by rijil-tr — APPROVED** [review]
LGTM!

**Pull request merged**